### ISSUE TITLE
Task add a checklist

### ISSUE THREAD

**Issue by joelkahan** [espocrm/espocrm#951]
Into the task should be able to add a checklist and show progress bar also show priority color each task

**Comment by m-hcb** [comment]
A task itself is a todo. However, you can create one to many relation of task entity with task entity and add subtasks under a task where you can set priority and status of subtasks.

**Comment by joelkahan** [comment]
why create subtasks, just add an option to make a checklist what to do in a single task,  and set the priority colors on the task for better recognize, like all task management software Trello Asana,

**Comment by beppe9000** [comment]
It would be cool to have it as a field type

**Comment by yurikuzn** [comment]
Checklist field type is panned.

**Comment by tomaspar** [comment]
When we can expect it?

**Comment by bandtank** [comment]
Any updates on this? A checklist would be very useful. I will help develop it if there's anything I can contribute.

**Comment by yurikuzn** [comment]
No updates yet. No enough time available.

**Comment by telecastg** [comment]
Here's a free plug-in for a checklist field 

https://github.com/telecastg/dynamic-checklist-for-espocrm

**Comment by bandtank** [comment]
@telecastg I haven't used the checklist type in the latest releases, so I'm not sure how your plugin is different. Could you briefly explain what your plugin does differently?

**Comment by telecastg** [comment]
Here's the thread in the forum, I think this will explain it.

Basically the out of the box "checklist" field type is a fixed list that you store in metadata, just like a regular enum field, but the plug-in allows you to edit that list on the fly and it is stored as an array of json objects in the database.

**Comment by bandtank** [comment]
That's amazing. Thank you for sharing this.

**Comment by telecastg** [comment]
You're very welcome, hopefully this will inspire other beneficiaries of the awesome EspoCRM platform to share their code and benefit everyone.

**Comment by rodrigoscoelho** [comment]
Great, congrats for helping with new features.

Maybe espo team merge it on the official code.


Enviado utilizando um Galaxy S10...

Em qui, 6 de fev de 2020 21:56, telecastg <notifications@github.com>
escreveu:

**Comment by telecastg** [comment]
Thanks, have a few more goodies that we implemented for our own use but have released as free extensions :-)

1) Enhanced Dynamic Logic : Allows you to set CSS for any field based on dynamic logic constrains and to use current user parameters (isAdmin, Role, Team, isPortal) as dynamic logic constrains, so for example you can hide certain field for members of a given Team.
https://forum.espocrm.com/forum/feature-requests/54547-enhanced-dynamic-logic

2) Electronic signature documents: Allows you to create and present documents for electronic signature.
https://forum.espocrm.com/forum/developer-help/54812-use-espocrm-to-create-render-and-print-documents-with-electronic-signatures

3) Enum (Select) field that is dynamically populated by SQL statements saved in metadata
https://forum.espocrm.com/forum/developer-help/55473-populate-enum-field-from-database-using-sql-statements-specified-in-metadata

4) Enum (Select) field that is dynamically populated by the options set for a different entity
https://forum.espocrm.com/forum/developer-help/55533-populate-enum-select-field-with-metadata-from-an-unrelated-entity

Please notice that the above extensions have not been extensively tested, nor are they paid extensions with any support, they are released more in the spirit of sharing experimental code to contribute to the EspoCRM platform growth so you might want to test them and check the code to get a better understanding of how they were implemented.

Saludos

**Comment by bandtank** [comment]
@telecastg These are awesome and very useful. Thank you for sharing. Do you have the code hosted anywhere? If so and if it's available for PRs, I would be happy to help fix any bugs I encounter or contribute to your extensions.

**Comment by telecastg** [comment]
Thanks @bandtank, yes I uploaded them to GitHub https://github.com/telecastg I would love to have contributors.

I am not a professional developer so I'm sure that my code can use all the help it can get.

**Comment by bandtank** [comment]
I will submit pull requests to your repository with the source code shown explicitly. I saw that you had uploaded the zip files to your repositories, which is very helpful, but it will be even better if the code is shown in the commits instead of only zip files. I will work on this shortly.

**Comment by beppe9000** [comment]
> I am not a professional developer so I'm sure that my code can use all the help it can get.

We could create a github organization for community extensions and learning resources.

**Comment by bandtank** [comment]
Good idea. I created an organization and invited a few people it. I also added the modules from @telecastg to get things started.

**Comment by joelkahan** [comment]
Share with us please the link, thank you so much

**Comment by bandtank** [comment]
Duh. Sorry.

https://github.com/EspoCRM-Custom-Modules

**Comment by joelkahan** [comment]
Thank you so much

**Comment by beppe9000** [comment]
> 
> Good idea @beppe9000. I created an organization and invited a few people it. I also added the modules from @telecastg to get things started.
>
> https://github.com/EspoCRM-Custom-Modules


Great! Be sure to post that in the forums too so we can properly rally the community 👍

**Comment by telecastg** [comment]
Excellent everyone !, I am just learning how to work with github, your contributions and advise are greatly appreciated ,
